addons/website_sale_stock/models/product_product.py

```diff
@@ -1,19 +1,49 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-from odoo import models
+from odoo import models, fields, _
 from odoo.http import request
 
 
 class ProductProduct(models.Model):
     _inherit = 'product.product'
 
+    stock_notification_partner_ids = fields.Many2many('res.partner', string='Back in stock Notifications')
+
+    def has_stock_notification(self, partner_id):
```

Comment:
I find it easier to understand if recordsets in variables do not end with `_id`/`_ids`, that way we can directly know whether it is a recordset or an id.
```suggestion
    def has_stock_notification(self, partner):
```